features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/presequite/SystemRequirementConfiguration.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.presequite
+
+internal data class SystemRequirementConfiguration internal constructor(
+    val cpuCoreNumber: Int,
+    val memorySizeInMb: Int
+) {
+
+    class Builder {
+        private var cpuCoreNumber: Int = 0
+        private var memorySize: Int = 0
+
+        fun setCpuCoreNumber(cpuCoreNumber: Int): Builder {
+            this.cpuCoreNumber = cpuCoreNumber
+            return this
+        }
+
+        fun setMemorySize(memorySize: Int): Builder {
+            this.memorySize = memorySize
+            return this
+        }
+
+        fun build(): SystemRequirementConfiguration {
+            return SystemRequirementConfiguration(
+                cpuCoreNumber = cpuCoreNumber,
+                memorySizeInMb = memorySize
+            )
+        }
+    }
+
+    companion object {
+        val DEFAULT = Builder()
+            .setMemorySize(256)
+            .setCpuCoreNumber(1)
```

Comment:
won't this never trigger for cpu because there is always at least 1 cpu core?

Reply:
The idea is to set a quite low requirement at first, to have less impact on current usage, then we will raise it depending on our benchmark